src/drt/src/MakeTritonRoute.cpp

```diff
@@ -50,11 +65,26 @@ void deleteTritonRoute(drt::TritonRoute* router)
 
 void initTritonRoute(OpenRoad* openroad)
 {
-  openroad->getTritonRoute()->init(openroad->tclInterp(),
-                                   openroad->getDb(),
-                                   openroad->getLogger(),
-                                   openroad->getDistributed(),
-                                   openroad->getSteinerTreeBuilder());
+  // Define swig TCL commands.
+  auto tcl_interp = openroad->tclInterp();
+  Drt_Init(tcl_interp);
+  sta::evalTclInit(tcl_interp, sta::drt_tcl_inits);
+
+  drt::TritonRoute* router = openroad->getTritonRoute();
+
+  std::unique_ptr<drt::AbstractGraphicsFactory> graphics_factory
+      = std::make_unique<drt::GraphicsFactory>();
+
+  std::unique_ptr<drt::ORDBInterface> or_db_interface
+      = std::make_unique<drt::ORDBInterface>();
+
+  router->init(openroad->getDb(),
+               openroad->getLogger(),
+               openroad->getDistributed(),
+               openroad->getSteinerTreeBuilder(),
+               std::move(graphics_factory),
+               std::move(or_db_interface),
+               ord::OpenRoad::openRoad()->getThreadCount());
```

Comment:
this may be modified by the user after initialization, but before calling for routing. This cannot be set here.